mermithid/processors/TritiumSpectrum/DistortedTritiumSpectrumLikelihoodSampler.py

```diff
@@ -0,0 +1,239 @@
+                c3 = ROOT.RooRealVar("c3", "c3", self.channel_eff_coeff[3])
+                c4 = ROOT.RooRealVar("c4", "c4", self.channel_eff_coeff[4])
+                cf = ROOT.RooRealVar("cf", "cf", self.channel_cf*1e-6-50)
+                channel_eff_coeff = ROOT.RooArgList(var, c0, c1, c2, c3, c4, cf)
+
+                # filter = args[4] * np.sqrt(1/(1 + 1*(x/args[0])**args[1]))* np.sqrt(1/(1 + 1*(x/args[2])**args[3]))
+                # x is Frequency in channel in MHz: (@0*TMath::Power(10, -6)-cf)
+                channelEffFunc = ROOT.RooFormulaVar("channel_efficiency", "channel_efficiency", "c4 * TMath::Sqrt(1/(1 + 1*TMath::Power((@0*TMath::Power(10, -6)-cf)/c0, c1)))* TMath::Sqrt(1/(1 + TMath::Power((@0*TMath::Power(10, -6)-cf)/c2, c3)))", channel_eff_coeff)
+                superDistortedSpectrum = ROOT.RooEffProd("superDistortedSpectrum", "superDistortedSpectrum", distortedSpectrum, channelEffFunc)
+                pdf = pdffactory.AddBackground(ROOT.RooAbsPdf)("pdf",var,superDistortedSpectrum,NEvents,NBkgd)
+
+            else:
+                pdf = pdffactory.AddBackground(ROOT.RooAbsPdf)("pdf",var,distortedSpectrum,NEvents,NBkgd)
+        else:
+            pdf = pdffactory.AddBackground(ROOT.RooAbsPdf)("pdf",var,shapePdf,NEvents,NBkgd)
+
+
+
+        # can = ROOT.TCanvas("can","can",600,400)
+        # frame = var.frame()
+        # totalSpectrum.plotOn(frame)
+        # frame.Draw()
+        # can.SaveAs("plots/model.pdf")
+
+        # Save pdf: this will save all required variables and functions
+
+        getattr(wspace,'import')(pdf)
+        return wspace
+
+    """
+    def _Generator(self):
+        '''
+        Generate the data by sampling the pdf defined in the workspace
+        '''
+        print("my own generator")
+        wspace = ROOT.RooWorkspace()
+        wspace = self.definePdf(wspace)
+        logger.debug("Workspace content:")
+        wspace.Print()
+        wspace = self._FixParams(wspace)
+        print("get pdf from wspace")
+        pdf = wspace.pdf("pdf")
+        print("getting params of interest", self.paramOfInterestNames)
+        paramOfInterest = self._getArgSet(wspace, self.paramOfInterestNames)
+        print("generate data")
+        data = pdf.generate(paramOfInterest, self.iter)
+        data.Print()
+
+        self.data = {}
+        for name in self.paramOfInterestNames:
+            self.data.update({name: []})
+
+        for i in range(0, data.numEntries()):
+            for item in self.data:
+                self.data[item].append(
+                    data.get(i).getRealValue(item))
+        self.data.update({"is_sample": [1]*(self.iter)})
+        return True
+
+    def _getArgSet(self, wspace, listNames):
+        argSet = ROOT.RooArgSet()
+        for name in listNames:
+            print(name)
+            argSet.add(wspace.var(name))
+        return argSet
+        """
+    def Frequency(self, E, B=None):
```

Comment:
These are useful functions (also defined in mermithid/processors/misc/TritiumAndEfficiencyBinner.py); they should be moved in a common place to avoid duplication.
Note also that these don't require the self, so they don't have to live into a processor or a class...

Reply:
moved